File: app/src/main/java/com/example/domain/model/AccountType.kt
package com.example.domain.model

enum class NormalBalance {
    DEBIT,
    CREDIT
}

enum class AccountType(
    val displayName: String,
    val normalBalance: NormalBalance,
    val defaultPrefix: String
) {
    ASSET("Asset", NormalBalance.DEBIT, "1"),
    LIABILITY("Liability", NormalBalance.CREDIT, "2"),
    EQUITY("Equity", NormalBalance.CREDIT, "3"),
    REVENUE("Revenue", NormalBalance.CREDIT, "4"),
    EXPENSE("Expense", NormalBalance.DEBIT, "5");

    companion object {
        fun fromString(value: String): AccountType {
            return entries.firstOrNull { 
                it.name.equals(value, ignoreCase = true) || it.displayName.equals(value, ignoreCase = true) 
            } ?: ASSET
        }
    }
}

File: backend/src/Accounting.Domain/Entities/AccountType.cs
namespace Accounting.Domain.Entities;

public enum AccountType
{
    Asset = 1,
    Liability = 2,
    Equity = 3,
    Revenue = 4,
    Expense = 5
}

public enum NormalBalance
{
    Debit = 1,
    Credit = 2
}

public static class AccountTypeExtensions
{
    public static NormalBalance GetNormalBalance(this AccountType type) => type switch
    {
        AccountType.Asset => NormalBalance.Debit,
        AccountType.Expense => NormalBalance.Debit,
        AccountType.Liability => NormalBalance.Credit,
        AccountType.Equity => NormalBalance.Credit,
        AccountType.Revenue => NormalBalance.Credit,
        _ => NormalBalance.Debit
    };
}
